compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java

```diff
@@ -103,7 +104,7 @@ public Symbol definition() {
         if (referredType.tag == TypeTags.PARAMETERIZED_TYPE || bType.tag == TypeTags.PARAMETERIZED_TYPE) {
             this.definition = symbolFactory.getBCompiledSymbol(((BParameterizedType) this.tSymbol.type).paramSymbol,
                                                                this.name());
-        } else if (referredType.tag == TypeTags.INTERSECTION) {
+        } else if (referredType.tag == TypeTags.INTERSECTION || referredType.tsymbol.origin == SymbolOrigin.VIRTUAL) {
```

Comment:
Instead of this approach, can't we check whether readonly flag is set or not?

Reply:
It is not clear at which position I should make your change. If you are suggesting replacing the virtual check with readonly, then it breaks the existing cases.

For instance, in the following case, the definition is correct as we are looking for the symbol through its name. However, since this symbol is also readonly, with the mentioned change, it would look for the tsymbol instead, which will produce an incorrect type

```ballerina
type Student string;

function fn() {
    Student student = "A";
}
```

Reply:
Anyway here, we have to make this change since we should avoid looking for virtual symbols.